src/components/Menus/Sidebar/DownloadPanel/DownloadPanel.test.js

```diff
@@ -0,0 +1,84 @@
+import DownloadPanel from "./DownloadPanel";
+import { fireEvent, render, waitFor, screen } from "@testing-library/react";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router";
+import configureStore from "redux-mock-store";
+import FileSaver from "file-saver";
+
+jest.mock("file-saver");
+jest.mock("jszip");
+jest.mock("jszip-utils", () => ({
+  getBinaryContent: jest.fn(),
+}));
+
+let container;
+describe("DownloadPanel", () => {
+  beforeEach(() => {
+    const middlewares = [];
+    const mockStore = configureStore(middlewares);
+    const initialState = {
+      editor: {
+        project: {
+          name: "My epic project",
+          identifier: "hello-world-project",
+          components: [
+            {
+              name: "main",
+              extension: "py",
+              content: "print('hello world')",
+            },
+          ],
+          image_list: [
+            {
+              url: "a.com/b",
+            },
+          ],
+        },
+      },
+    };
+    const store = mockStore(initialState);
+
+    container = render(
+      <Provider store={store}>
+        <MemoryRouter>
+          <DownloadPanel />
+        </MemoryRouter>
+      </Provider>,
+    );
+  });
+  test("Renders the correct heading", () => {
+    expect(container.getByText("downloadPanel.heading")).not.toBeNull();
```

Comment:
More recently I've tended to switch these over from `not.toBeNull` to `toBeInTheDocument`, but they're probably more or less doing the same thing, so it's up to you

Reply:
Yes - agreed. better to be specific!